ffi/src/lib.rs

```diff
@@ -121,6 +121,14 @@ impl KernelStringSlice {
             len: source.len(),
         }
     }
+
+    /// Create a new empty `KernelStringSlice`. `len` will be 0, and `ptr` will be `null`.
+    pub(crate) fn empty() -> Self {
+        Self {
+            ptr: std::ptr::null(),
```

Comment:
The `TryFromSlice` docs specifically state:
> The slice must be a valid (non-null) pointer

... which is required because we're going to turn the pointer into a reference, and a null reference is UB even if never accessed. We should really use `NonNull` and its associated `dangling()` instead -- see e.g. `KernelBoolSlice`). Tho I suspect that `ptr: NonNull::from("")` would work just as well for this specific case.

Reply:
Ahh, good catch! I've actually just removed this and we can just use `""` since it avoids all the mess